plugins/action/git_retrieve.py

```diff
@@ -178,9 +178,15 @@ def _clone(self: T) -> None:
             command_parts.extend(cli_parameters)
             no_log[token_base64] = "<TOKEN>"
 
-        command_parts.extend(
-            ["clone", "--depth=1", "--progress", "--no-single-branch", origin],
-        )
+        tag = self._task.args["origin"].get("tag")
+        if tag:
+            command_parts.extend(
+                ["clone", "--depth=1", "--progress", "--branch", tag, origin],
```

Comment:
The addition of clone, depth, and progress could be done outside the conditional since they are not unique to the condition